### PR TITLE
enhance: optimize milvus core building

### PR DESCRIPTION
**Pull request by chyezh**
issue: #35549,#35611,#35633

- remove milvus_segcore milvus_indexbuilder..., add libmilvus_core
- core building only link once
- move opendal compilation into cmake
- fix odr

### CODE REVIEW

File: internal/core/src/common/ChunkFileWriter.cpp [hunk 1]
```diff
@@ -1,35 +0,0 @@
-// Copyright (C) 2019-2020 Zilliz. All rights reserved.
```
Comment (PwzXxm) [inline internal/core/src/common/ChunkFileWriter.cpp:1]: Confirmed by @sunby. Intentionally deleted.

Comment (chyezh) [reply internal/core/src/common/ChunkFileWriter.cpp:1]: Already confirmed.

**Pull request merged**